src/game/boe.newgraph.cpp

```diff
@@ -869,8 +869,14 @@ std::string get_item_interesting_string(cItem item) {
 		case eItemType::HELM:
 		case eItemType::GLOVES:
 		case eItemType::SHIELD_2:
-		case eItemType::BOOTS: // TODO: Verify that this is displayed correctly
-			sout << "Blocks " << item.item_level + ((item.protection > 0) ? 1 : 0) << '-' << item.item_level + item.protection << " damage";
+		case eItemType::BOOTS: // coherent with damage_pc
+			sout << "Blocks "
+			<< (item.item_level>0 ? 1 : item.item_level) + (item.bonus>0 ? 1+item.bonus/2 : item.bonus) +
+				(item.protection > 0 ? 1 : item.protection)
```

Comment:
This new formula looks extremely weird. Can you point to the exact lines of code that it's "coherent" with?

In particular, the `item_level>0 ? 1 : item_level` bit doesn't seem right.

Reply:
I tried to convert the damage_pc code to show the maximum damage protection:
```
		for (i = 0; i < 24; i++)
			if ((adven[which_pc].items[i].variety != 0) && (adven[which_pc].equip[i] == TRUE)) {
				if ((adven[which_pc].items[i].variety >= 12) && (adven[which_pc].items[i].variety <= 17)) {
						r1 = get_ran(1,1,adven[which_pc].items[i].item_level);
						how_much -= r1;
						
						// bonus for magical items
						if (adven[which_pc].items[i].bonus > 0) {
							r1 = get_ran(1,1,adven[which_pc].items[i].bonus);
							how_much -= r1;
							how_much -= adven[which_pc].items[i].bonus / 2;
							}
						if (adven[which_pc].items[i].bonus < 0) {
							how_much = how_much - adven[which_pc].items[i].bonus;
							}
						r1 = get_ran(1,0,100);
						if (r1 < hit_chance[adven[which_pc].skills[8]] - 20)
							how_much -= 1;
					}
				if (adven[which_pc].items[i].protection > 0) {
						r1 = get_ran(1,1,adven[which_pc].items[i].protection);
						how_much -= r1;
					}
				if (adven[which_pc].items[i].protection < 0) {
						r1 = get_ran(1,1,-1 * adven[which_pc].items[i].protection);
						how_much += r1;
					}
				}	

```
but yes, I make some mistakes,  

- concerning item_level : if (item_level<=0) we add a number in [item_level:1] else [1, item_level]
- concerning the bonus : if (bonus<0) we add bonus else [1+bonus/2, 3*bonus/2]
- concerning the protection : if (protection<0) we add [protection,-1] else [1,protection]

Reply:
I think this should be correct, right?

68ef066dca17823ddf9b99f2de9c6e0902b0d6bf

Reply:
> Okay, so is your `item_level <= 0` check based on the fact that `get_ran` returns `times * min` if `max` is less than `min`? (That would also happen for `item_level == 0` though.)

Yes.

> 
> If 'bonus' and 'protection' are negative but 'item_level' is still positive, you get the following formula:
> 
> [1, item_level] - bonus - [1, -protection]
> 

This is [1, item_level] + bonus - [1, -protection]
( ie. we add -bonus damage, so the protection is negative : bonus )

> I think this should be correct, right?
> 
> [68ef066](https://github.com/calref/cboe/commit/68ef066dca17823ddf9b99f2de9c6e0902b0d6bf)

This seems to be correct. I don't know if there is enough room to display the result; if so, it might be better to display "minprotect - maxprotect" or "minprotect—maxproctect" to avoid displaying -3--2.